extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/taskadapter/K8sTaskAdapterTest.java

```diff
@@ -272,6 +275,123 @@ public PodSpec getSpec()
     );
   }
 
+
+  @Test
+  public void fromTask_doesnt_pass_loadLookups_Argument() throws IOException
+  {
+    final PodSpec podSpec = K8sTestUtils.getDummyPodSpec();
+    TestKubernetesClient testClient = new TestKubernetesClient(client)
+    {
+      @SuppressWarnings("unchecked")
+      @Override
+      public <T> T executeRequest(KubernetesExecutor<T> executor) throws KubernetesResourceNotFoundException
+      {
+        return (T) new Pod()
+        {
+          @Override
+          public PodSpec getSpec()
+          {
+            return podSpec;
+          }
+        };
+      }
+    };
+
+    KubernetesTaskRunnerConfig config = KubernetesTaskRunnerConfig.builder()
+                                                                  .withNamespace("test")
+                                                                  .build();
+    K8sTaskAdapter adapter = new SingleContainerTaskAdapter(
+        testClient,
+        config,
+        taskConfig,
+        startupLoggingConfig,
+        node,
+        jsonMapper,
+        taskLogs
+    );
+    Task task = new NoopTask(
+        "id",
+        "id",
+        "datasource",
+        0,
+        0,
+        ImmutableMap.of("context", RandomStringUtils.randomAlphanumeric((int) DruidK8sConstants.MAX_ENV_VARIABLE_KBS * 20))
+    );
+    Job job = adapter.fromTask(task);
+    //Verify that --loadLookups NONE shouldn't be passed
+    Set<String> arguments = Arrays.stream(job.getSpec()
+                                           .getTemplate()
+                                           .getSpec()
+                                           .getContainers()
+                                           .get(0)
+                                           .getArgs()
+                                           .get(0).split(" ")).collect(Collectors.toSet());
+
+    Assert.assertFalse(arguments.containsAll(ImmutableSet.of("--loadLookups", LookupLoadingMode.NONE.name())));
+  }
+
+
+  @Test
+  public void fromTask_passes_loadLookups_Argument() throws IOException
```

Comment:
Seems like a lot of code can be commoned out between these two tests. You can either merge these into a single test where task1 passes the argument and task2 doesn't, or use a common private method that is called by both the tests.

Reply:
refactored to private method